### PR TITLE
feat: [FW-13] add home page

### PR DESCRIPTION
**Pull request by barroro**
- Add hero section
- Add how it works section
- Add animations using [framer-motion](https://www.framer.com/motion/)
- Add eslint rule for unused vars
- Change CI file, now it's using just one file to run all checks instead
    - Use Make to run checks
- Add husky for pre-commit

### CODE REVIEW

**Comment by DanielMoraDC** [comment]
Could we have separate PRs for the 3 points in the description. So reviewing is much simpler and changes are smaller 😄

File: .github/workflows/.github-ci.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+      - name: Install dependencies
+        run: npm install
+
+      # Step to run tests
+      - name: Run tests
+        run: npm test
+
+      # Step to run linter (ESLint and Prettier)
+      - name: Run linter
+        run: npm run lint
```
Comment (DanielMoraDC) [inline .github/workflows/.github-ci.yml:41]: The main advantage of pre-commit is that they "don't let you" push if your changes break the linting rules defined. Now we need to wait until the tests run in CI, which is less convenient. Could be combine those?

File: .github/workflows/.github-ci.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        with:
+          node-version: ${{ matrix.node-version }}
+
+      # Install dependencies
+      - name: Install dependencies
+        run: npm install
+
+      # Step to run tests
+      - name: Run tests
+        run: npm test
```
Comment (DanielMoraDC) [inline .github/workflows/.github-ci.yml:37]: Could we use `make` for the steps here, so we can easily recreat them locally?

File: .github/workflows/.github-ci.yml (hunk 1 above)
Comment (barroro) [reply .github/workflows/.github-ci.yml:41]: We can cover this problem using the husky library, running automatically tests and eslint before push each commit.
Check package: [Husky docu](https://typicode.github.io/husky/)

File: .github/workflows/.github-ci.yml (hunk 2 above)
Comment (barroro) [reply .github/workflows/.github-ci.yml]: Done! 👌

File: src/pages/home/components/HowItWorksSection/HowItWorksSection.tsx [hunk 3]
```diff
@@ -0,0 +1,84 @@
+import { motion } from "framer-motion"
+import Heading from "@/components/ui/Heading"
+import customizePlaylistImage from "@public/customize-playlist.svg"
+import spotifyLockImage from "@public/spotify-lock.svg"
+import searchArtistsImage from "@public/search-artists.svg"
+import AnimatedSection from "./AnimatedSection"
+
+const containerVariants = {
```
Comment (DanielMoraDC) [inline src/pages/home/components/HowItWorksSection/HowItWorksSection.tsx:8]: I see `containerVariants` and `itemVariants` are repeated across files. Should we reuse them?

File: .github/workflows/.github-ci.yml [hunk 4]
```diff
@@ -0,0 +1,37 @@
+name: Node.js CI
+
+# Run on every push or pull request to the 'main' branch (you can change this to your preference)
```
Comment (DanielMoraDC) [inline .github/workflows/.github-ci.yml:3]: I feel the comments in this file are redundant. I would rather remove them.

File: .github/workflows/.github-ci.yml [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    branches:
+      - main
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [18.x]  # You can adjust the Node.js version according to your project
```
Comment (DanielMoraDC) [inline .github/workflows/.github-ci.yml:18]: The `nvmrc` file points to node 20, should we also use that here for consistency?

File: README.md [hunk 6]
```diff
@@ -17,7 +14,6 @@ make local-setup
 
 This will install pre-commit hooks, the npm dependencies and prepare the env file. Make sure to fill the env file with the corresponding variables. You will need to to create your own Spotify app following [these instructions](https://developer.spotify.com/documentation/web-api/tutorials/getting-started#create-an-app). Use `http://<host>:3000/api/auth/callback/spotify` as a redirect URI in the app configurtation (we are using `localhost` as host, for now).
```
Comment (DanielMoraDC) [inline README.md:15]: We should update this part accordingly (pre-commit hooks).

File: src/pages/home/components/HowItWorksSection/HowItWorksSection.tsx [hunk 7]
```diff
@@ -0,0 +1,84 @@
+import { motion } from "framer-motion"
```
Comment (DanielMoraDC) [inline src/pages/home/components/HowItWorksSection/HowItWorksSection.tsx:1]: This section appears on the main page but there is also a link to `Hot it Works` in the header 🤔

File: src/pages/home/index.tsx [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+          content="Spotify playlists Generator using festival line-up"
+        />
+        <meta
+          name="keywords"
+          content="spotify, generator, playlist, festival, line-up"
+        />
+        <meta property="og:title" content="Home" />
+        <meta
+          property="og:description"
+          content="Descripción para redes sociales."
```
Comment (DanielMoraDC) [inline src/pages/home/index.tsx:21]: This description and the url, image and twitter ones seem to have placeholder values. Should we update them?

File: .github/workflows/.github-ci.yml (hunk 5 above)
Comment (barroro) [reply .github/workflows/.github-ci.yml:18]: It's true, I change it!

File: src/pages/home/components/HowItWorksSection/HowItWorksSection.tsx (hunk 7 above)
Comment (barroro) [reply src/pages/home/components/HowItWorksSection/HowItWorksSection.tsx:1]: Yes, I think the page "How it works" doesn't make sense. We could remove the page and keep the section on the home page.

File: src/pages/home/index.tsx (hunk 8 above)
Comment (barroro) [reply src/pages/home/index.tsx:21]: I have kept this metadata to update in the next iterations, but I can remove it and add it if we need to in the future.

File: src/pages/home/components/HowItWorksSection/HowItWorksSection.tsx (hunk 3 above)
Comment (barroro) [reply src/pages/home/components/HowItWorksSection/HowItWorksSection.tsx:8]: I think it has different values inside, but I can think of reusing this code.

File: src/pages/home/index.tsx (hunk 8 above)
Comment (DanielMoraDC) [reply src/pages/home/index.tsx:21]: I'd rather remove it and add back when needed.

File: .github/workflows/.github-ci.yml [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+    branches:
+      - main
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [20.9.0]  # You can adjust the Node.js version according to your project
```
Comment (DanielMoraDC) [inline .github/workflows/.github-ci.yml:17]: Can we also remove the rest of the comments in this file? They feel a bit redundant, too.

Comment (barroro) [reply .github/workflows/.github-ci.yml]: done!

File: src/pages/home/components/HowItWorksSection/HowItWorksSection.tsx (hunk 3 above)
Comment (barroro) [reply src/pages/home/components/HowItWorksSection/HowItWorksSection.tsx]: I have created a shared file for the framer motion variants in the lib folder. It looks great 👌

**Review by DanielMoraDC — APPROVED** [review]

**Pull request merged**